src/Service/Resolvers/CosmosClientProvider.cs

```diff
@@ -58,21 +60,29 @@ private void InitializeClient(RuntimeConfig? configuration)
 
             if (string.IsNullOrEmpty(_connectionString) || configuration.ConnectionString != _connectionString)
             {
+                string userAgent = GetCosmosUserAgent();
+                CosmosClientOptions options = new()
+                {
+                    ApplicationName = userAgent
+                };
+
                 _connectionString = configuration.ConnectionString;
                 ParseCosmosConnectionString();
 
                 if (!string.IsNullOrEmpty(_accountKey))
                 {
-                    Client = new CosmosClientBuilder(_connectionString).WithContentResponseOnWrite(true).Build();
+                    Client = new CosmosClientBuilder(_connectionString).WithContentResponseOnWrite(true)
+                        .WithApplicationName(userAgent)
```

Comment:
By setting this app name, how is it discovered in production? Will the logs spit out the appname?

Trying to understand how will we benefit from this change..

Reply:
It shows up in cosmos telemetry